src/api/portfolioFunctions.js

```diff
@@ -0,0 +1,103 @@
+import { SafePromiseAll } from './helpers'
+import { goatLowestPrice } from './dataSources'
+import createRequestObject from './createRequest'
+
+export async function getPortfolioData (userID, currency, location) {
+  try {
+    const request = createRequestObject('portfolio_get', {
+      user_id: userID,
+      currency
+    })
+    const response = await fetch(request.url, request.headers)
+
+    let portfolio = await response.json()
+    portfolio.forEach(item => { item.data = JSON.parse(item.data) })
+
+    // Get current prices for each item
+    const reqs = portfolio.map(item => itemWithCurrentPrice(
+      item, item.size, item.gender, currency, location
+    ))
+    portfolio = await SafePromiseAll(reqs)
+
+    return portfolio
+  } catch (err) {
+    console.log(err)
+    return []
+  }
+}
+
+export async function addToPortfolio (data) {
+  try {
+    const request = createRequestObject('portfolio_add', data)
+    const response = await fetch(request.url, request.headers)
+    return await response.text()
+  } catch (err) {
+    console.log(err)
+  }
+}
+
+export async function removeFromPortfolio (recordId) {
+  try {
+    const request = createRequestObject('portfolio_remove', {
+      record_id: recordId
+    })
+    return fetch(request.url, request.headers)
```

Comment:
`remove` returns the whole response but `add` just takes the text - we should make that consistent

Reply:
returning the promise in `remove` is necessary because it needs to be resolved downstream due to a very weird bug. but returning `response.text()` in `add` is unnecessary so I can change that.